src/reference/informative/informC04_component_reference.rst

```diff
@@ -8,4 +8,189 @@
+            <component name="JerryLeadbetter" component_ref="husband" />
+          </import>
+
+        </model>
+
+        <!-- Inside the GoodFamily.cellml file: -->
+        <model name="TheGoodFamilyUnit">
+          <component name="husband" />
+          <component name="wife" />
+        </model>
+
+        <!-- Inside the LeadbetterFamily.cellml file: -->
+        <model name="TheLeadbetterFamilyUnit">
+          <component name="husband" />
+          <component name="wife" />
+        </model>
+
+    Now let's make things more interesting by adding an encapsulation hierarchy:
+
+    .. code::
+
+      model: SurbitonNeighbourhood
+        ├─ component: WomenInTheNeighbourhood
+        │   ├─ component: BarbaraGood <╴╴╴╴╴╴╴╴╴╴╴┐
+        │   └─ component: MargotLeadbetter <╴╴╴╴┐ ╷
+        └─ component: MenInTheNeighbourhood     ╷ ╷
+            ├─ component: TomGood <╴╴╴╴╴╴╴╴╴╴╴┐ ╷ ╷
+            └─ component: JerryLeadbetter <╴┐ ╷ ╷ ╷
+                                            ╷ ╷ ╷ ╷
+                                    imported components
+                                            ╵ ╵ ╵ ╵
+      # In LeadbetterFamily.cellml:         ╵ ╵ ╵ ╵
+      model: LeadbetterFamilyModel          ╵ ╵ ╵ ╵
+        ├─ component: husband ╴╴╴╴╴╴╴╴╴╴╴╴╴╴┘ ╵ ╵ ╵
+        └─ component: wife ╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴┼╴┘ ╵
+                                              ╵   ╵
+      # In GoodFamily.cellml:                 ╵   ╵
+      model: GoodFamilyModel                  ╵   ╵
+        ├─ component: husband ╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴┘   ╵
+        └─ component: wife ╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴┘
+
+    .. container:: toggle
+
+      .. container:: header
+
+        See CellML syntax
+
+      .. code-block:: xml
+
+        <model name="SurbitonNeighbourhood">
+          ...  
+          <component name="WomenInTheNeighbourhood" />
+          <component name="MenInTheNeighbourhood" />
+
+          <!-- Throughout the importing model (i.e.: this model), the imported
+              items are referred to by their local name attribute ("BarbaraGood" etc), 
+              not the name they are called within their imported family units ("wife"). -->
+          <encapsulation>
+            <component_ref component="WomenInTheNeighbourhood">
+              <component_ref component="BarbaraGood" />
+              <component_ref component="MargotLeadbetter" />
+            </component_ref>
+            <component_ref component="MenInTheNeighbourhood">
+              <component_ref component="TomGood" />
+              <component_ref component="JerryLeadbetter" />
+            </component_ref>
+          </encapsulation>
+        </model>
+
+    This particular encapsulation structure means that the women (Barbara and Margot) are essentially unable to have any contact with the men (Tom and Jerry) even though their original components in the models from which they were imported *were* able to access one another.
```

Comment:
Interesting! This is clearly valid CellML, but I wonder whether this couldn't result in a non-sensical model. I mean, what if `TomGood` and `BarbaraGood` have connected variables in `GoodFamily.cellml`?

Reply:
You tell me ... :) ?  I don't think that connections and mapped variables are imported too? (otherwise the entire imported model would probably come along for the ride ...) And in any case, the two components are totally separate in the importing model, so I'd have thought that no automatic connection was possible ...?

Reply:
The only person who has ever done any advanced things with imports is @nickerso, so I will leave it to him to comment on this... :)

Reply:
I think I had a conversation with @hsorby about this ... can you confirm what the expected behaviour is, Hugh?

Reply:
Only connections internal to a component hierarchy are kept therefore any external connections are lost.

Reply:
This makes sense, and my comment still stands in that we could end up with something non-sensical, but it's "fine".

Reply:
OK, thanks both.  I'll add a comment about that.